static/js/dashboard-spa.js

```diff
@@ -0,0 +1,538 @@
+// only allow lowercase letters, numbers, dots (.), dashes (-) and underscores (_)
+// don't allow dot at the start or end or consecutive dots
+const ALIAS_PREFIX_REGEX = /^(?!\.)(?!.*\.$)(?!.*\.\.)[0-9a-z-_.]+$/;
+
+new Vue({
+  el: '#dashboard-app',
+  delimiters: ["[[", "]]"], // necessary to avoid conflict with jinja
+  data: {
+    showFilter: false,
+    showStats: false,
+
+    mailboxes: [],
+
+    // variables for creating alias
+    canCreateAlias: true,
+    isLoading: true,
+    aliasPrefixInput: "",
+    aliasPrefixError: "",
+    aliasSuffixes: [],
+    aliasSelectedSignedSuffix: "",
+    aliasNoteInput: "",
+    defaultMailboxId: "",
+
+    // variables for aliases list
+    isFetchingAlias: true,
+    aliasesArray: [], // array of existing alias
+    aliasesArrayOfNextPage: [], // to know there is a next page if not empty
+    page: 0,
+    isLoadingMoreAliases: false,
+    searchString: "",
+    filter: "", // TODO add more filters and also sorting when backend API is ready
+  },
+  computed: {
+    isLastPage: function () {
+      return this.aliasesArrayOfNextPage.length === 0;
+    },
+  },
+  async mounted() {
+
+    if (store.get("showFilter")) {
+      this.showFilter = true;
+    }
+
+    if (store.get("showStats")) {
+      this.showStats = true;
+    }
+
+    await this.loadInitialData();
+
+  },
+  methods: {
+    // initialize mailboxes and alias options and aliases
+    async loadInitialData() {
+      this.isLoading = true;
+      await this.loadMailboxes();
+      await this.loadAliasOptions();
+      this.isLoading = false;
+      await this.loadAliases();
+    },
+
+    async loadMailboxes() {
+      try {
+        const res = await fetch("/api/mailboxes");
+        if (res.ok) {
+          const result = await res.json();
+          this.mailboxes = result.mailboxes;
+          this.defaultMailboxId = this.mailboxes.find((mailbox) => mailbox.default).id;
+        } else {
+          throw new Error("Could not load mailboxes");
+        }
+      } catch (err) {
+        toastr.error("Sorry for the inconvenience! Could you try refreshing the page? ", "Could not load mailboxes");
+      }
+    },
+
+    async loadAliasOptions() {
+      this.isLoading = true;
+      try {
+        const res = await fetch("/api/v5/alias/options");
+        if (res.ok) {
+          const aliasOptions = await res.json();
+          this.aliasSuffixes = aliasOptions.suffixes;
+          this.aliasSelectedSignedSuffix = this.aliasSuffixes[0].signed_suffix;
```

Comment:
as this is expired after a short time, I think it's better to only load alias options when user clicks on the "new custom alias" button.